src/parameters_loader.rs

```diff
@@ -19,8 +19,66 @@ pub struct ParametersValues {
 }
 define_global_property!(Parameters, ParametersValues);
 
+/// Check whether parameters provided in input json
+/// are valid using our knowledge of allowable parameter
+/// values. This prevents us from, say, loading the synth
+/// pop only to realize that we accidentally set the GI to
+/// be a negative number, which would cause a panic much
+/// later in the code (when transmission happens).
+fn validate(parameters: &ParametersValues) -> Result<(), IxaError> {
+    if parameters.r_0 < 0.0 {
+        return Err(IxaError::IxaError(
+            "r_0 must be a non-negative number".to_string(),
+        ));
+    }
+    // need to think about a better way of validation
+    // in the long term that makes sense for an arbitrary GI
+    if parameters.generation_interval <= 0.0 {
+        return Err(IxaError::IxaError(
+            "r_0 must be a non-negative number".to_string(),
+        ));
+    }
+    Ok(())
+}
+
 pub fn init_parameters(context: &mut Context, file_path: &Path) -> Result<(), IxaError> {
     let parameters_json = context.load_parameters_from_json::<ParametersValues>(file_path)?;
+    validate(&parameters_json)?;
     context.set_global_property_value(Parameters, parameters_json);
     Ok(())
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use crate::parameters_loader::ParametersValues;
+
+    #[test]
+    fn test_validate_r_0() {
+        let parameters = ParametersValues {
+            population: 1000,
```

Comment:
We are not using population

Reply:
Yes, agree -- I requested this to be removed in the population loader PR as that makes the most thematic sense, so I think we need to land that first. I will rebase this to main once we land that.